torchaudio/transforms/_transforms.py

```diff
@@ -2089,3 +2089,58 @@ def forward(
 
         specgram_enhanced.to(dtype)
         return specgram_enhanced
+
+
+class SoudenMVDR(torch.nn.Module):
+    r"""Minimum Variance Distortionless Response (*MVDR* [:footcite:`capon1969high`]) module
+    based on the method proposed by *Souden et, al.* [:footcite:`souden2009optimal`].
+
+    .. devices:: CPU CUDA
+
+    .. properties:: Autograd TorchScript
+
+    .. math::
+        \textbf{w}_{\text{MVDR}}(f) =
+        \frac{{{\bf{\Phi}_{\textbf{NN}}^{-1}}(f){\bf{\Phi}_{\textbf{SS}}}}(f)}
+        {\text{Trace}({{{\bf{\Phi}_{\textbf{NN}}^{-1}}(f) \bf{\Phi}_{\textbf{SS}}}(f))}}\bm{u}
+
+    where :math:`\bf{\Phi}_{\textbf{SS}}` and :math:`\bf{\Phi}_{\textbf{NN}}`
+    are the power spectral density (PSD) matrices of speech and noise, respectively.
+    :math:`\bf{u}` is a one-hot vector that represents the reference channel.
+    """
+
+    def forward(
+        self,
+        specgram: Tensor,
+        psd_s: Tensor,
+        psd_n: Tensor,
+        reference_channel: Union[int, Tensor],
+        diagonal_loading: bool = True,
+        diag_eps: float = 1e-7,
+        eps: float = 1e-8,
+    ) -> Tensor:
+        """
+        Args:
+            specgram (Tensor): Multi-channel complex-valued spectrum.
+                Tensor of dimension `(..., channel, freq, time)`
+            psd_s (Tensor): The complex-valued power spectral density (PSD) matrix of target speech.
+                Tensor of dimension `(..., freq, channel, channel)`
```

Comment:
is there a check somewhere that enforces equality between the last two dimensions?

Reply:
Good point. Adding such check helps users better understand the module usage.

Reply:
This check is applicable to `F.rtf_power`, `F.rtf_evd`, `F.mvdr_weights_rtf`, `F.mvdr_weights_souden`. It's better to add it in a separate PR.

Reply:
sure sounds good